idl/chromadb/proto/query_executor.proto

```diff
@@ -6,9 +6,9 @@ import "chromadb/proto/chroma.proto";
 
 message ScanOperator {
     Collection collection = 1;
-    string knn_id = 2;
-    string metadata_id = 3;
-    string record_id = 4;
+    Segment knn = 2;
```

Comment:
will this have version compatibility issues? For e.g. what if FE that is on a newer version that has this code sends a query rpc to Query that is on an older version without these changes? And vice-versa

Reply:
I've made this backward compatible